src/Components/Blueprints/BlueprintsSideBar.tsx

```diff
@@ -49,27 +47,13 @@ type emptyBlueprintStateProps = {
   bodyText: string;
 };
 
-const BlueprintsSidebar = ({
-  selectedBlueprint,
-  setSelectedBlueprint,
-}: blueprintProps) => {
-  const [blueprintsSearchQuery, setBlueprintsSearchQuery] = useState<
-    string | undefined
-  >();
-  const debouncedSearch = useCallback(
-    debounce((filter) => {
-      setBlueprintsSearchQuery(filter.length > 0 ? filter : undefined);
-    }, 300),
-    [setBlueprintsSearchQuery]
-  );
-  React.useEffect(() => {
-    return () => {
-      debouncedSearch.cancel();
-    };
-  }, [debouncedSearch]);
+const BlueprintsSidebar = () => {
+  const selectedBlueprint = useAppSelector(selectSelectedBlueprint);
+  const blueprintsSearchQuery = useAppSelector(selectBlueprintSearchInput);
```

Comment:
Would you mind changing this to blueprintSearchInput? I find it easier to understand the code when the name of the variable in the React component is the same as it is in the store.